### DISCUSSION TITLE
Automatically disable VRR temporarily when a specific window is in focus

### DISCUSSION THREAD

**Discussion by Atemo-C** [hyprwm/Hyprland#5315] [Q&A]
Greetings.

Due to some Electron shenanigans, and maybe some NVIDIA ones *(though unlikely)*, I encounter pretty bad flickering when variable refresh rate is enabled whilst using Electron programs, such as FreeTube as an example. This is **not** an issue of Hyprland nor wlroots, as far as I know, which is why I am posting this in Discussions rather than in Issues.

To somewhat get around this problem for now, I currently have my custom program launcher disable VRR when launching an Electron program, and it gets re-enabled when the program is closed. However, this is not ideal, as I may want to have VRR enabled when said Electron program is not my main focus.
I also have a keyboard shortcut and toggle in my status bar to toggle VRR on/off *(all my programs behave perfectly when toggling VRR on/off, even repeatedly)*.

My question, therefore, is this: Is there a way to configure Hyprland so that when a specific window *(class/title)* is currently visible on the monitor *(I only have one display)*, VRR automatically gets disabled, and when said window is no longer appearing on-screen, VRR gets enabled back automatically?

If no, would there be any way or even interest to implement this into Hyprland due to how niche this issue is?
If yes, is it somewhere in the documentation that I may have missed?

Thanks in advance for any help. :fox_face:

**Comment by adamnejm** [comment]
Maybe we can elevate this to an issue? Mentioning people that were previously interested in #7621 
All you have to do is thumbs up the original discussion post.

CC: @heywatsuphello @Gliczy @8Vinso8 @noxvail @BluewyDiamond @mtroberts @Atemu @sononio

**Comment by Gliczy** [comment]
This is no longer a problem since the addition of `vrr=3` and the windowrule for setting content types.

You can set `vrr=3` to only enable vrr for full-screened windows with content type `game` or `video`.
And you can use a windowrule to set the content type.
For games running with proton this would be: `content game, ^(steam_app_.*)$`.

**Reply by adamnejm** [reply]
OK, looks like #9334 didn't update the Wiki to include the `vrr` variable changes. [I PR'd it just now](https://github.com/hyprwm/hyprland-wiki/pull/1039)
It's still rather stupid. I essentially want to enable VRR for everything besides Firefox, Alacritty and MPV, so I had to implement this:
```sh
windowrule = content game, class:.*
windowrule = content none, class:^(Alacritty|firefox|mpv)$
vrr = 3
```

This straight up looks like abusing a content type, but also may cause some issues in the future if the `game` or `video` content controls more things than only VRR OOTB or if I want to change behavior depending on those types.

INB4 Why not set the content type for games only? Well, I don't want to upkeep a long ass list of (native/non-proton/non-umu) games that I play or might play in the future. Terraria, Left 4 Dead, Team Fortress, Valheim, etc. See where I'm going with this?
One way this could work if games communicate via Wayland and request to be treated like a game, but that's a fool's dream.

I don't want to underestimate the addition of `vrr=3`, it's decent, but I still think a dedicated window rule in addition to that would be nice.

**Comment by vaxerski** [comment]
This is reasonable. Promoted https://github.com/hyprwm/Hyprland/issues/10147
